src/Compilers/VisualBasic/Portable/Scanner/CharacterInfo.vb

```diff
@@ -257,7 +257,19 @@ Namespace Microsoft.CodeAnalysis.VisualBasic
 
         Friend Shared Function BeginsBaseLiteral(c As Char) As Boolean
             Return (c = "H"c Or c = "O"c Or c = "B"c Or c = "h"c Or c = "o"c Or c = "b"c) OrElse
-                    (IsFullWidth(c) AndAlso (c = FULLWIDTH_LATIN_CAPITAL_LETTER_H Or c = FULLWIDTH_LATIN_CAPITAL_LETTER_O Or c = FULLWIDTH_LATIN_CAPITAL_LETTER_B Or c = FULLWIDTH_LATIN_SMALL_LETTER_H Or c = FULLWIDTH_LATIN_SMALL_LETTER_O Or c = FULLWIDTH_LATIN_CAPITAL_LETTER_B))
+                    (IsFullWidth(c) AndAlso (BeginsBaseLiteral_Hex(c) OrElse BeginsBaseLiteral_Oct(c) OrElse BeginsBaseLiteral_Bin(c)))
+        End Function
+
+        Private Shared Function BeginsBaseLiteral_Hex(c As Char) As Boolean
+            Return c = FULLWIDTH_LATIN_CAPITAL_LETTER_H Or c = FULLWIDTH_LATIN_SMALL_LETTER_H
```

Comment:
`OrElse`, here and in other locations.

Reply:
@cston I left the expressions as close to originals as possible, as this method is called a lot. Using `OrElse` would likely make the unable to be JIT-Inlined at it increase the IL size for each usage case. The Fullwidth variants are not as common as their half-width cousins.
Inlining the, makes the already long expression even longer, the functions help. *Even potentially JIT Inline able.

Reply:
Sorry, I missed the fact that the original code was using `Or` rather than `OrElse`. Regarding inlining, the original was inlined, and it looks like the fix is to the last term in that expression. Given that, please consider simply changing the last term in the original expression.